File: backend/app/schemas/media_intel.py
from datetime import datetime
from enum import Enum
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class WatchKind(str, Enum):
    PULSE = "pulse"
    COMPETITOR = "competitor"
    TREND = "trend"
    BRAND = "brand"


class BriefRequest(BaseModel):
    topic: str = Field(..., min_length=2, max_length=200)
    kind: WatchKind = WatchKind.PULSE


class WatchResponse(BaseModel):
    id: str
    topic: str
    kind: WatchKind
    last_brief: Optional[Dict[str, Any]] = None
    last_run_at: Optional[datetime] = None
    workspace_id: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class WatchList(BaseModel):
    items: List[WatchResponse]


class BriefResponse(BaseModel):
    brief: Dict[str, Any]

File: frontend/lib/api/mediaIntel.ts
import { http } from "./http";

export type WatchKind = "pulse" | "competitor" | "trend" | "brand";

export interface BriefDevelopment {
  title: string;
  detail: string;
  source_url?: string;
  source_name?: string;
  recency?: string;
}

export interface BriefSource {
  title: string;
  url: string;
  source?: string;
  published?: string | null;
}

export interface Brief {
  topic: string;
  kind: WatchKind;
  headline: string;
  summary: string;
  key_developments: BriefDevelopment[];
  sentiment: string;
  opportunities: string[];
  risks?: string[];
  sources: BriefSource[];
  provider: string;
  generated_at: string;
}

export interface Watch {
  id: string;
  topic: string;
  kind: WatchKind;
  last_brief: Brief | null;
  last_run_at: string | null;
  workspace_id: string;
  created_at: string;
  updated_at: string;
}

export const mediaIntelApi = {
  brief: (topic: string, kind: WatchKind) =>
    http.post<{ brief: Brief }>("/media-intel/brief", { topic, kind }),
  listWatches: () => http.get<{ items: Watch[] }>("/media-intel/watchlist"),
  addWatch: (topic: string, kind: WatchKind) =>
    http.post<Watch>("/media-intel/watchlist", { topic, kind }),
  refreshWatch: (id: string) => http.post<Watch>(`/media-intel/watchlist/${id}/refresh`),
  deleteWatch: (id: string) => http.del<void>(`/media-intel/watchlist/${id}`),
};
